e2e_test.go

```diff
@@ -118,11 +118,27 @@ func runIntegrationTest(t *testing.T, testFile string) {
 			}
 
 			filePath := filepath.Join(dir, tt.Give)
+			// diff case
+			if tt.WantDiff != nil {
+				args := append([]string{"-d", filePath}, args...)
+				var stderr bytes.Buffer
+				var stdout bytes.Buffer
+				require.NoError(t, run(args, bytes.NewReader(stdin), &stderr, &stdout),
+					"could not run patch")
+				want := strings.Split(string(tt.WantDiff), "\n")
+				got := strings.Split(stdout.String(), "\n")
+				l1 := len(want)
+				l2 := len(got)
+				for i := range want {
+					assert.Equal(t, want[l1-1-i], got[l2-1-i])
```

Comment:
I believe this loop is unnecessary, and you can just check for `assert.Equal(t, want, got)`. Also, it should not require splitting those strings.

Reply:
It was not possible to add a filepath section in the testdata --diff-- section. A temp directory is created in the e2e_test and the filepath changes everytime.  . Hence, I was checking if the diff output sans filepath matched to the one in testdata --diff-- section.
For example, in the output : 
```
--- /var/folders/6k/pdyxq4md59j30sl9cdrhn1p80000gn/T/gopatch-integration3493611168/time.go
+++ /var/folders/6k/pdyxq4md59j30sl9cdrhn1p80000gn/T/gopatch-integration3493611168/time.go
@@ -1,7 +1,7 @@
 package sender
 
 // Send does stuff.
-func Send(string) (*Response, error) {
+func Send(context.Context, string) (*Response, error) {
 	fmt.Println("Sending request")
 	// Here's a comment.
 	return &Response{}, nil
```
Just checking the diff & ignoring the 
```
--- /var/folders/6k/pdyxq4md59j30sl9cdrhn1p80000gn/T/gopatch-integration3493611168/time.go 
+++ /var/folders/6k/pdyxq4md59j30sl9cdrhn1p80000gn/T/gopatch-integration3493611168/time.go
```
check as the test was to check the diff output.
I'm checking the whole diff output in testPreview in main_test.go

I could also manually add the filepath to the wantDiff?